sql/catalyst/src/main/java/org/apache/spark/sql/connector/write/LogicalWriteInfo.java

```diff
@@ -45,4 +45,14 @@ public interface LogicalWriteInfo {
    * the schema of the input data from Spark to data source.
    */
   StructType schema();
+
+  /**
+   * the schema of the input metadata from Spark to data source.
+   */
+  StructType metadataSchema();
```

Comment:
I could add default implementations and return null but I think instances of this interface are only created by Spark.